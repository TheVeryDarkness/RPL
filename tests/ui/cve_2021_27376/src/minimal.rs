fn main() {
    let s: *const core::net::SocketAddrV4 = core::ptr::null();
    let t1 = s as *const libc::sockaddr;
    //~^ ERROR: wrong assumption of layout compatibility from `std::net::SocketAddrV4` to `libc::sockaddr`
    let s: *const core::net::SocketAddrV6 = core::ptr::null();
    let t2 = s as *const libc::sockaddr;
    //~^ ERROR: wrong assumption of layout compatibility from `std::net::SocketAddrV6` to `libc::sockaddr`
}
