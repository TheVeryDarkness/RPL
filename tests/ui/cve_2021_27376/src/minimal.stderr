error: wrong assumption of layout compatibility from `std::net::SocketAddrV6` to `libc::sockaddr`
  --> tests/ui/cve_2021_27376/src/minimal.rs:6:14
   |
LL |     let t2 = s as *const libc::sockaddr;
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: it's not guaranteed by Rust standard library. See https://github.com/rust-lang/rust/pull/78802
note: casted from this
  --> tests/ui/cve_2021_27376/src/minimal.rs:6:14
   |
LL |     let t2 = s as *const libc::sockaddr;
   |              ^

error: wrong assumption of layout compatibility from `std::net::SocketAddrV4` to `libc::sockaddr`
  --> tests/ui/cve_2021_27376/src/minimal.rs:3:14
   |
LL |     let t1 = s as *const libc::sockaddr;
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: it's not guaranteed by Rust standard library. See https://github.com/rust-lang/rust/pull/78802
note: casted from this
  --> tests/ui/cve_2021_27376/src/minimal.rs:3:14
   |
LL |     let t1 = s as *const libc::sockaddr;
   |              ^

error: aborting due to 2 previous errors

